main.js

```diff
@@ -294,6 +324,36 @@ export class IdDecoder {
         bitLength: this.fixedBitLength
       });
     }
+    if(this.multihash) {
+      // <varint hash fn code>: identity function
+      const [hashFnCode] = decoded;
+
+      if(hashFnCode !== MULTIHASH_IDENTITY_FUNCTION_CODE) {
+        throw new Error('Invalid multihash function code.');
+      }
+      // <varint digest size in bytes>
+      const digestSize = decoded[1];
+
+      if(digestSize > 127) {
+        throw new Error(`Decoded identifier size too large.`);
+      }
+
+      const bytes = decoded.subarray(2);
+
+      if(this.expectedSize) {
+        if(digestSize !== this.expectedSize) {
+          throw new Error('Unexpected identifier size.');
+        }
+
+        if(bytes.byteLength !== this.expectedSize) {
+          throw new Error(
+            `Invalid decoded identifier size. Identifier must be ` +
+              `"${this.expectedSize}" bytes.`);
+        }
+      }
```

Comment:
Making the expectedSize check optional adds another edge case where the size might never get checked at all.  How about refactoring it so the multihash size is always checked against the actual size, but the expectedSize is only checked if needed.

Also use RangeError for size checks.

```suggestion
      const bytes = decoded.subarray(2);

      if(bytes.byteLength !== digestSize) {
        throw new RangeError('Unexpected identifier size.');
      }

      if(this.expectedSize && bytes.byteLength !== this.expectedSize) {
        throw new RangeError(
          `Invalid decoded identifier size. Identifier must be ` +
            `"${this.expectedSize}" bytes.`);
      }
```

Reply:
Thanks, updated fcd0ad9